src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -306,7 +315,7 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
                 }
 
                 // In order to open documents in VS we need to understand the link from temp file to document and its encoding etc.
-                _fileToDocumentInfoMap[info.FilePath] = new(documentId, encoding, sourceProject.Id, sourceWorkspace);
+                _fileToDocumentInfoMap[info.FilePath] = new(documentId, encoding, info.ChecksumAlgorithm, sourceProject.Id, sourceWorkspace);
```

Comment:
Is there a reason we're passing in checksum algorithsm for each file but the encoding just once? I'd imagine they behave the same, right?

Reply:
Default encoding is stored in the PDB for the entire compilation. Documents in the same compilation can't have different default encoding.

Reply:
@tmat So this is just an artifact that PDBs can (theoretically) have different checksums per document?

Reply:
Yes. It's not just theory. They definitely do. You can specify checksum for a document via `#pragma checksum`.